From bc3515bd83d324dbda60aa22b0527ceeafe7b1c0 Mon Sep 17 00:00:00 2001
From: Pardha Kesapragada <pkesapragada@habana.ai>
Date: Thu, 18 Feb 2021 16:24:56 +0200
Subject: [PATCH] [SW-20154] make: specify files individually

If files to be compiled are specified using a wildcard
(like *) then the other target files will also be compiled.
This will create a problem, because fuse read function
is having a strong definition in onix.c, which will
always be generated into the final image. Hence, add
C files individually.

Change-Id: Id60eaba8b0af7fc484f44acca75fa05af66e84f4
Signed-off-by: Pardha Kesapragada <pkesapragada@habana.ai>
---

diff --git a/Makefile b/Makefile
index 2cade4d..8117136 100644
--- a/Makefile
+++ b/Makefile
@@ -43,7 +43,10 @@
 include $(HL_DEV)/Makefile
 
 ARCBOOT_COMMON_SRCS_C := $(shell ls common/*.c common/rsa/*.c $(HL_DEV)/$(HL_TARGET).c)
-ARCBOOT_COMMON_SRCS_C += $(shell ls $(HL_DEV)/*.c)
+ARCBOOT_COMMON_SRCS_C += $(shell ls $(HL_DEV)/security.c)
+ARCBOOT_COMMON_SRCS_C += $(shell ls $(HL_DEV)/common.c)
+ARCBOOT_COMMON_SRCS_C += $(shell ls $(HL_DEV)/secureboot.c)
+
 
 ARCBOOT_RAM_SRCS_C := app/arcboot_ram.c
 ARCBOOT_RAM_SRCS_S := app/crt0_ram.S
